cli/src/main/java/com/devonfw/tools/ide/process/ProcessResultImpl.java

```diff
@@ -23,23 +24,25 @@ public class ProcessResultImpl implements ProcessResult {
 
   private final List<String> err;
 
+  private final List<OutputMessage> outputMessages;
+
   /**
    * The constructor.
    *
    * @param executable the {@link #getExecutable() executable}.
    * @param command the {@link #getCommand() command}.
    * @param exitCode the {@link #getExitCode() exit code}.
-   * @param out the {@link #getOut() out}.
-   * @param err the {@link #getErr() err}.
+   * @param outputMessages {@link #getOutputMessages() output Messages}.
    */
-  public ProcessResultImpl(String executable, String command, int exitCode, List<String> out, List<String> err) {
+  public ProcessResultImpl(String executable, String command, int exitCode, List<OutputMessage> outputMessages) {
 
     super();
     this.executable = executable;
     this.command = command;
     this.exitCode = exitCode;
-    this.out = Objects.requireNonNullElse(out, Collections.emptyList());
-    this.err = Objects.requireNonNullElse(err, Collections.emptyList());
+    this.outputMessages = Objects.requireNonNullElse(outputMessages, Collections.emptyList());
+    this.out = this.outputMessages.stream().filter(outputMessage -> !outputMessage.error()).map(OutputMessage::message).collect(Collectors.toList());
+    this.err = this.outputMessages.stream().filter(OutputMessage::error).map(OutputMessage::message).collect(Collectors.toList());
```

Comment:
Small simplification:
```suggestion
    this.out = this.outputMessages.stream().filter(outputMessage -> !outputMessage.error()).map(OutputMessage::message).toList();
    this.err = this.outputMessages.stream().filter(OutputMessage::error).map(OutputMessage::message).toList();
```

Reply:
done